_source/_posts/2021-01-20-reactive-java-microservices.adoc

```diff
@@ -0,0 +1,545 @@
+---
+layout: blog_post
+title: "Reactive Java Microservices with Spring Boot and JHipster"
+author: matt-raible
+by: advocate
+communities: [java]
+description: "Learn how to create a reactive microservices architecture using Spring Boot and JHipster."
+tags: [java, reactive, microservices, spring-boot, kotlin, jhipster]
+tweets:
+- "Learn how to build a reactive @java microservices architecture using @springboot and @jhipster in this quick tutorial from @mraible."
+- "Reactive #Java microservices are easy to create with @jhipster! Check out our 2021 guide. 👇"
+- "Want to build reactive Java microservices? @jhipster, @springboot, and @springcloud to the rescue! ⛑"
+image: blog/reactive-java-microservices/reactive-java-microservices.png
+type: conversion
+---
+:toc: macro
+:page-liquid:
+:experimental:
+
+Java has been at the forefront of microservice architectures since they came to prominence a few years ago. It's a popular language with well-known, high-quality frameworks, like Spring Boot, Spring Cloud, Spring Data, and Spring Security.
+
+Spring Boot 2.0 introduced a new web framework called Spring WebFlux. Before, Spring Boot only shipped with Spring MVC as an option. WebFlux offers a way for developers to do _reactive programming_. This means you can write your code with familiar syntax and, as a result, your app will use fewer resources and scale better.
+
+image::{% asset_path 'blog/reactive-java-microservices/diagram-boot-reactor.svg' %}[alt=Spring Boot 2.0,width=700,align=center]
```

Comment:
Great image!

Reply:
I agree. It's no longer on spring.io, so I had to use the wayback machine to get it.